### PR TITLE
en, zh: explain TidbCluster spec.discovery

### PR DESCRIPTION
**Pull request by better0332**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-tidb-operator) that's required for repo owners to accept my contribution.

### What is changed, added, or deleted? (Required)

 explan TidbCluster `spec.discovery`

### Which TiDB Operator version(s) do your changes apply to? (Required)

- [*] master (the latest development version)
- [x] v1.1 (TiDB Operator 1.1 versions)
- [ ] v1.0 (TiDB Operator 1.0 versions)

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [*] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: zh/configure-a-tidb-cluster.md [hunk 1]
```diff
@@ -161,6 +161,31 @@ TiDB Operator 支持为 PD、TiDB、TiKV 挂载多块 PV，可以用于不同用
...
+    requests:
+      cpu: "0.2"
+  pd:
+    baseImage: pingcap/pd
+    replicas: 1
+    requests:
+      storage: "1Gi"
+    config: {}
+...
+```
```
Comment (DanielZhangQD) [inline zh/configure-a-tidb-cluster.md:187]:
```suggestion

```yaml
apiVersion: pingcap.com/v1alpha1
kind: TidbCluster
metadata:
  name: basic
spec:
  version: v5.0.0
  pvReclaimPolicy: Retain
  discovery:
    limits:
      cpu: "0.2"
    requests:
      cpu: "0.2"
  pd:
    baseImage: pingcap/pd
    replicas: 1
    requests:
      storage: "1Gi"
    config: {}
...
```
```

**Comment by qiancai** [comment]
Hi @better0332, please confirm if the same change is applicable for v1.2 as well. If yes, please select the v1.1 (TiDB Operator 1.1 versions) checkbox in the PR description. Thanks!

File: zh/configure-a-tidb-cluster.md [hunk 2]
```diff
@@ -161,6 +161,32 @@ TiDB Operator 支持为 PD、TiDB、TiKV 挂载多块 PV，可以用于不同用
 
 PD、TiKV、TiDB、TiFlash、TiCDC 及 Pump 支持配置 Pod 使用宿主机上的网络命名空间 [`HostNetwork`](https://kubernetes.io/docs/concepts/policy/pod-security-policy/#host-namespaces)。可通过配置 `spec.hostNetwork: true` 为所有受支持的组件开启，或通过为特定组件配置 `hostNetwork: true` 为单个或多个组件开启。
 
+### Discovery
+
+TiDB Operator 会为每一个 TiDB 集群启动一个 Discovery 服务，Discovery 服务会为每个 PD Pod 返回相应的启动参数，来辅助 PD 集群启动。可以通过 `spec.discovery` 配置 Discovery 服务的资源，详见[容器资源管理](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/)。
```
Comment (qiancai) [inline zh/configure-a-tidb-cluster.md:166]:
```suggestion
TiDB Operator 会为每一个 TiDB 集群启动一个 Discovery 服务。Discovery 服务会为每个 PD Pod 返回相应的启动参数，来辅助 PD 集群启动。你可以通过 `spec.discovery` 配置 Discovery 服务的资源，详见[容器资源管理](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/)。
```

File: en/configure-a-tidb-cluster.md [hunk 3]
```diff
@@ -170,6 +170,32 @@ To enable `HostNetwork` for all supported components, configure `spec.hostNetwor
 
 To enable `HostNetwork` for specified components, configure `hostNetwork: true` for the components.
 
+### Discovery
+
+TiDB Operator will start a Discovery service for each TiDB cluster, the Discovery service will return the corresponding startup parameters for each PD Pod to assist it start. The resources of the Discovery service can be configured through `spec.discovery`, see for details [Managing Resources for Containers](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/)
```
Comment (qiancai) [inline en/configure-a-tidb-cluster.md:175]:
```suggestion
TiDB Operator starts a Discovery service for each TiDB cluster. The Discovery service can return the corresponding startup parameters for each PD Pod to support the startup of the PD cluster. You can configure resources of the Discovery service using `spec.discovery`. For details, see [Managing Resources for Containers](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/).
```

**Review by DanielZhangQD — APPROVED** [review]
LGTM

File: en/configure-a-tidb-cluster.md [hunk 4]
```diff
@@ -170,6 +170,32 @@ To enable `HostNetwork` for all supported components, configure `spec.hostNetwor
 
 To enable `HostNetwork` for specified components, configure `hostNetwork: true` for the components.
 
+### Discovery
+
+TiDB Operator starts a Discovery service for each TiDB cluster. The Discovery service can return the corresponding startup parameters for each PD Pod to support the startup of the PD cluster. You can configure resources of the Discovery service using `spec.discovery`. For details, see [Managing Resources for Containers](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/).
+
```
Comment (qiancai) [inline en/configure-a-tidb-cluster.md:176]:
```suggestion

A `spec.discovery` configuration example is as follows:

```

File: zh/configure-a-tidb-cluster.md [hunk 5]
```diff
@@ -161,6 +161,32 @@ TiDB Operator 支持为 PD、TiDB、TiKV 挂载多块 PV，可以用于不同用
 
 PD、TiKV、TiDB、TiFlash、TiCDC 及 Pump 支持配置 Pod 使用宿主机上的网络命名空间 [`HostNetwork`](https://kubernetes.io/docs/concepts/policy/pod-security-policy/#host-namespaces)。可通过配置 `spec.hostNetwork: true` 为所有受支持的组件开启，或通过为特定组件配置 `hostNetwork: true` 为单个或多个组件开启。
 
+### Discovery
+
+TiDB Operator 会为每一个 TiDB 集群启动一个 Discovery 服务。Discovery 服务会为每个 PD Pod 返回相应的启动参数，来辅助 PD 集群启动。你可以通过 `spec.discovery` 配置 Discovery 服务的资源，详见[容器资源管理](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/)。
+
```
Comment (qiancai) [inline zh/configure-a-tidb-cluster.md:167]:
```suggestion

`spec.discovery` 配置示例：

```

File: en/configure-a-tidb-cluster.md [hunk 6]
```diff
@@ -170,6 +170,32 @@ To enable `HostNetwork` for all supported components, configure `spec.hostNetwor
...
+```yaml
+apiVersion: pingcap.com/v1alpha1
+kind: TidbCluster
+metadata:
+  name: basic
+spec:
+  version: v5.0.0
```
Comment (qiancai) [inline en/configure-a-tidb-cluster.md:183]:
```suggestion
  version: v5.0.1
```

File: zh/configure-a-tidb-cluster.md [hunk 7]
```diff
@@ -161,6 +161,32 @@ TiDB Operator 支持为 PD、TiDB、TiKV 挂载多块 PV，可以用于不同用
...
+```yaml
+apiVersion: pingcap.com/v1alpha1
+kind: TidbCluster
+metadata:
+  name: basic
+spec:
+  version: v5.0.0
```
Comment (qiancai) [inline zh/configure-a-tidb-cluster.md:174]:
```suggestion
  version: v5.0.1
```

**Review by qiancai — APPROVED** [review]

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Pull request merged**